core/psh/nc.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Phoenix-RTOS
+ *
+ * nc - TCP and UDP connect or listen
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/select.h>
+
+#include "psh.h"
+
+
+/*
+ * This could be achieved using getaddrinfo(),
+ * however it does not recognize scope id for now
+ * and requires dynamic memory, which we
+ * are trying to avoid.
+ */
+static int psh_nc_sockaddrFill(int *af_family, char *addrstr, char *portstr,
+                               struct sockaddr_storage *sockaddr,
+                               socklen_t *addrlen)
+{
+	struct sockaddr_in6 *saddr_in6;
+	struct sockaddr_in *saddr;
+	char *ifname, *endptr;
+	int port = 0, scopeid = 0;
+	uint32_t addr[4] = { };
```

Comment:
What this empty initializer is for?

https://stackoverflow.com/questions/17589533/is-an-empty-initializer-list-valid-c-code

Reply:
My intention was to initialize this array with zeros. I didn't know, this syntax is C++ specific and not compliant with C99. I'll substitute it with ` = { 0 }`.